src/Microsoft.Diagnostics.EventFlow.Core/Implementations/Metadata/RequestData.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using Validation;
+
+namespace Microsoft.Diagnostics.EventFlow.Metadata
+{
+    public enum DurationUnit
+    {
+        TimeSpan,
+        Milliseconds,
+        Seconds,
+        Minutes,
+        Hours
+    };
+
+    public class RequestData
+    {
+        public static readonly string RequestMetadataKind = "request";
+        public static readonly string RequestNamePropertyMoniker = "requestNameProperty";
+        public static readonly string IsSuccessPropertyMoniker = "isSuccessProperty";
+        public static readonly string DurationPropertyMoniker = "durationProperty";
+        public static readonly string DurationUnitMoniker = "durationUnit";
+
+        public string RequestName { get; private set; }
+        public TimeSpan Duration { get; private set; }
+        public bool IsSuccess { get; private set; }
+
+        // Ensure that RequestData can only be created using TryGetRequestData() method
+        private RequestData() { }
+
+        public static DataRetrievalResult TryGetData(
+            EventData eventData,
+            EventMetadata requestMetadata,
+            out RequestData request)
+        {
+            Requires.NotNull(eventData, nameof(eventData));
+            Requires.NotNull(requestMetadata, nameof(requestMetadata));
+            request = null;
+
+            string requestNameProperty = requestMetadata[RequestNamePropertyMoniker];
+            if (string.IsNullOrWhiteSpace(requestNameProperty))
+            {
+                return DataRetrievalResult.MissingMetadataProperty(RequestNamePropertyMoniker);
+            }
+
+            string requestName = null;
+            if (!eventData.GetValueFromPayload<string>(requestNameProperty, (v) => requestName = v))
+            {
+                return DataRetrievalResult.DataMissingOrInvalid(requestNameProperty);
+            }
+
+            string isSuccessProperty = requestMetadata[IsSuccessPropertyMoniker];
+            if (string.IsNullOrWhiteSpace(isSuccessProperty))
```

Comment:
@yantang-msft this is a good suggestion, thanks, Will make isSuccess and duration optional.

A note on HTTP method/response code: these are only valid for HTTP-based requests, so cannot be mandatory. I will add a response code property as optional. Will keep it as a string so that it is as general as possible, covering non-HTTP scenarios too.